pkg/remote/remote_catalog.go

```diff
@@ -129,8 +132,8 @@ func (c *RemoteCatalog) createOrUpdateCatalog(ctx context.Context,
 	}
 
 	// it can happen that the ModuleTemplate CRD is not existing in the Remote Cluster when we apply it and retry
-	if containsMetaIsNoMatchErr(errs) {
-		if err := c.CreateModuleTemplateCRDInRuntime(ctx, v1beta2.ModuleTemplateKind.Plural()); err != nil {
+	if containsMetaIsNoMatchErr(errs) || len(errs) == 0 {
```

Comment:
now I get it, by this, you change the way the `CreateModuleTemplateCRDInRuntime` was originally designed for. It should only designed for create module template crd in remote skr, as the described, it should be really cautious when you change the existing function behavior, because it can introduce unexpected result, for example, if we have test for this function, then probably it will break.

Two options here:
1. rename this function to something like createRemoteCRDAndUpdateKymaAnnotation, but as you can see, this function did two much job, which violates the single responsibility principle. It's also hard for you to write a unit test for it. Which is not suggest way.
2. Put you update Kyma annotation in a different function, just for updating kyma annotation, and also please write a unit test for it.